.github/workflows/build.yml

```diff
@@ -115,174 +102,175 @@ jobs:
+          cmake -S . -B ${{ env.BUILD_DIR }} -DCMAKE_INSTALL_PREFIX=${{ env.INSTALL_DIR }} -DCMAKE_BUILD_TYPE=${{ inputs.build_type }} -G Ninja
 
-      - name: Configure CMake on Windows portable 
-        if: runner.os == 'Windows' && matrix.portable == true
-        shell: msys2 {0}
+      - name: Configure CMake (Linux)
+        if: runner.os == 'Linux'
         run: |
-          cmake -S . -B ${{ env.BUILD_DIR }} -DCMAKE_INSTALL_PREFIX=${{ env.INSTALL_DIR }} -DCMAKE_BUILD_TYPE=${{ inputs.build_type }} -G Ninja 
+          cmake -S . -B ${{ env.BUILD_DIR }} -DCMAKE_INSTALL_PREFIX=/usr -DCMAKE_BUILD_TYPE=${{ inputs.build_type }} -G Ninja
 
-      - name: Configure CMake on Linux
-        if: runner.os == 'Linux' && matrix.portable != true
-        run: |
-          cmake -S . -B ${{ env.BUILD_DIR }} -DCMAKE_INSTALL_PREFIX=/usr -DCMAKE_BUILD_TYPE=${{ inputs.build_type }} -DLauncher_PORTABLE=OFF -G Ninja
-
-      - name: Configure CMake on Linux Portable 
-        if: runner.os == 'Linux' && matrix.portable == true
-        run: |
-          cmake -S . -B ${{ env.BUILD_DIR }} -DCMAKE_INSTALL_PREFIX=${{ env.INSTALL_DIR }} -DCMAKE_BUILD_TYPE=${{ inputs.build_type }} -G Ninja
+      ##
+      # BUILD
+      ##
 
       - name: Build
         if: runner.os != 'Windows'
         run: |
           cmake --build ${{ env.BUILD_DIR }}
 
-      - name: Build on Windows
+      - name: Build (Windows)
         if: runner.os == 'Windows'
         shell: msys2 {0}
         run: |
           cmake --build ${{ env.BUILD_DIR }}
 
-      - name: Install
-        if: runner.os != 'Linux' && runner.os != 'Windows'
+      ##
+      # PACKAGE BUILDS
+      ##
+
+      - name: Package (macOS)
+        if: runner.os == 'macOS'
         run: |
           cmake --install ${{ env.BUILD_DIR }}
 
-      - name: Install on Windows
+          cd ${{ env.INSTALL_DIR }}
+          macdeployqt "PolyMC.app" -executable="PolyMC.app/Contents/MacOS/polymc" -always-overwrite
+          chmod +x "PolyMC.app/Contents/MacOS/polymc"
+          tar -czf ../PolyMC.tar.gz *
+
+      - name: Package (Windows)
         if: runner.os == 'Windows'
         shell: msys2 {0}
         run: |
           cmake --install ${{ env.BUILD_DIR }}
 
-      - name: Install on Linux
-        if: runner.os == 'Linux' && matrix.portable != true
+          cd ${{ env.INSTALL_DIR }}
+          if [ "${{ matrix.msystem }}" == "mingw32" ]; then
+            cp /mingw32/bin/libcrypto-1_1.dll /mingw32/bin/libssl-1_1.dll ./
+          elif [ "${{ matrix.msystem }}" == "mingw64" ]; then
+            cp /mingw64/bin/libcrypto-1_1-x64.dll /mingw64/bin/libssl-1_1-x64.dll ./
+          fi
+
+      - name: Package (Windows, portable)
+        if: runner.os == 'Windows'
+        shell: msys2 {0}
+        run: |
+          cmake --install ${{ env.BUILD_DIR }} --prefix ${{ env.INSTALL_PORTABLE_DIR }}
```

Comment:
i think here you should copy what you've done before but adding the portable stuff.
The current way installs two times, that makes the windows action even slower

Reply:
Sounds like a good idea. Forgot about the slow install step on Windows